### PR TITLE
[vividus] Enable dotall mode for step asserting against regex

### LINKED ISSUE #1967: Step Then `${textContent}` matches `.*text.*` does not work
**Describe the bug**
Body of mail has word "Dear". Step Then `${textContent}` matches `.*Dear.*` does not work. This step does not look for matches in the text

**To Reproduce**
Scenario: Fetch email message
When I fetch email message from `gmail` server filtered by
|parameter     |rule            |value                                       |
|SUBJECT       |EQUAL_TO        |Registration code                           |
|SENT_DATE     |GREATER_THAN    |1970-01-01T00:00:00.000Z                    |
|RECEIVED_DATE |LESS_THAN       |2020-06-14T11:47:11.521Z                    |
|FROM          |IS_EQUAL_TO     |service <regservice@company.com>            |
|TO_RECIPIENTS |EQUAL_TO        |test@company.com|
 and save message content to SCENARIO variable `textContent`
Then `${textContent[1]}` matches `.*Dear.*`

**Expected behavior**
Word 'Dear' is found in mail body

**Artifacts**
[Artifacts_Error_Allure.zip](https://github.com/vividus-framework/vividus/files/7178514/Artifacts_Error_Allure.zip)

**Execution environment (please complete the following information):**
 - Vividus version: 0.3.8
 - OS: MAcOS 11.4
 - Java: JDK 16

**Desktop Web (if applicable, please complete the following information):**
 - Browser Chrome

### PR DESCRIPTION
**Pull request by valfirst**
Fixes #1967

### CODE REVIEW

File: docs/modules/commons/pages/vividus-steps.adoc [hunk 1]
```diff
@@ -138,3 +138,17 @@ Given I initialize the scenario variable `personalInfo` using template `template
 |name     |race  |
 |Dagoth Ur|Dunmer|
 ----
+
+=== Assert the value matches regex
+
+Asserts that the given value matches the specified regular expression.
+The dotall mode is enabled by default: the expression `.` matches any character,
+including a line terminator.
```
Comment (uarlouski) [inline docs/modules/commons/pages/vividus-steps.adoc:146]: should not we also add that dot all matches ^ and $ only at the start and end of string?

File: vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java [hunk 2]
```diff
@@ -176,14 +176,15 @@ public void tablesAreEqual(Object variable, ExamplesTable table)
...
+     * @param regex The regular expression
      */
-    @Then("`$value` matches `$pattern`")
-    public void valueMatchesPattern(String value, Pattern pattern)
+    @Then("`$value` matches `$regex`")
+    public void assertMatchesRegex(String value, String regex)
     {
-        softAssert.assertThat(String.format("Value '%s' matches pattern '%s'", value, pattern.pattern()), value,
+        Pattern pattern = Pattern.compile(regex, Pattern.DOTALL);
+        softAssert.assertThat(String.format("Value '%s' matches pattern '%s'", value, regex), value,
```
Comment (uarlouski) [inline vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java:187]: are we going to change pattern to regex in the future?

**Review by ikalinin1 — APPROVED** [review]

File: vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java (hunk 2 above)
Comment (valfirst) [reply vividus/src/main/java/org/vividus/bdd/steps/BddVariableSteps.java:187]: probably, but not for now, it will lead to the breaking changes

**Review by abudevich — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

File: docs/modules/commons/pages/vividus-steps.adoc (hunk 1 above)
Comment (valfirst) [reply docs/modules/commons/pages/vividus-steps.adoc:146]: `^` and `$` match start/end of line online in conjunction with MULTILINE flag

**Pull request merged**